src/components/homepage/calculator-preview.tsx

```diff
@@ -0,0 +1,367 @@
+import { useEffect, useState } from "react";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { Card, CardContent } from "@/components/ui/card";
+import { Input } from "../ui/input";
+import { Switch } from "../ui/switch";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { CALCULATOR_CONSTANTS } from "@/lib/calculatorConstants";
+import { Button } from "../ui/button";
+import { motion } from "framer-motion";
+
+const minWage = CALCULATOR_CONSTANTS.MIN_WAGE;
+const nonTaxableAmountOfTaxBasis =
+  CALCULATOR_CONSTANTS.NON_TAXABLE_AMOUNT_OF_TAX_BASIS;
+const incomeTax = CALCULATOR_CONSTANTS.INCOME_TAX;
```

Comment:
Tieto konštanty čo si tu iba premenovávaš - to tak nerob. Páči sa mi že máš tie calculator related konštanty zgrupené pod jedným objektom, ale tu si to znova premenovávaš aby si to v jednom objekte nemal. Keď ti vadí že máš ten názov potom dlhý (CALCULATOR_CONSTANTS.NON_TAXABLE_AMOUNT...), tak to vieš pošéfit premenovaním importu 

`import { CALCULATOR_CONSTANTS as CC } from "@/lib/calculatorConstants";`

a potom všade v kóde používať 

`CC.MIN_WAGE`